webapp/views/App/views/Analysis/Chain/store/actions/index.js

```diff
@@ -16,19 +16,36 @@ export const useActions = ({
   setDirty,
   step,
   setStep,
+  stepDirty,
+  setStepDirty,
+  setStepOriginal,
   calculation,
   setCalculation,
+  calculationDirty,
+  setCalculationDirty,
+  setCalculationOriginal,
 }) => ({
-  onInit: useOnInit({ chain, setChain, step, setStep, calculation, setCalculation }),
-  onDismiss: useOnDismiss({ chain, setChain, dirty, setDirty }),
-  onSave: useOnSave({ chain, setChain, step, calculation }),
+  init: useInit({ chain, setChain, step, setStep, setStepDirty, calculation, setCalculation, setCalculationDirty }),
+  dismiss: useDismiss({ chain, setChain, dirty, setDirty }),
+  save: useSave({
+    chain,
+    setChain,
+    step,
+    calculation,
+    stepDirty,
+    setStepDirty,
+    setStepOriginal,
+    calculationDirty,
+    setCalculationDirty,
+    setCalculationOriginal,
+  }),
   canSelectNodeDef: useCanSelectNodeDef({ chain }),
   getAttributeUuidsOtherChains: useGetAttributeUuidsOtherChains({
     attributesUuidsOtherChains,
     setAtrributesUuidsOtherChains,
     chain,
   }),
-  addEntityVirtual: useAddEntityVirtual(),
-  addNodeDefAnalysis: useAddNodeDefAnalysis({ step, calculation }),
+  addEntityVirtual: useAddEntityVirtual({ chain, step, stepDirty, calculation, calculationDirty }),
+  addNodeDefAnalysis: useAddNodeDefAnalysis({ chain, step, stepDirty, calculation, calculationDirty }),
```

Comment:
the code would be much more readable if we follow our convention.
these actions should take directly the state object

Reply:
on it